tests/qxmppmixinvitation/tst_qxmppmixinvitation.cpp

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (C) 2008-2020 The QXmpp developers
+ *
+ * Author:
+ *  Melvin Keskin
+ *
+ * Source:
+ *  https://github.com/qxmpp-project/qxmpp
+ *
+ * This file is a part of QXmpp library.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ */
+
+#include "QXmppMixInvitation.h"
+
+#include "util.h"
+#include <QDomDocument>
+#include <QObject>
+
+class tst_QXmppMixInvitation : public QObject
+{
+    Q_OBJECT
+
+private slots:
+    void testInvitation();
+    void testIsInvitation();
+};
+
+void tst_QXmppMixInvitation::testInvitation()
+{
+    const QByteArray xml(
+        "<invitation xmlns=\"urn:xmpp:mix:misc:0\">"
+            "<inviter>hag66@shakespeare.example</inviter>"
+            "<invitee>cat@shakespeare.example</invitee>"
+            "<channel>coven@mix.shakespeare.example</channel>"
+            "<token>ABCDEF</token>"
+        "</invitation>"
+    );
+
+    QXmppMixInvitation invitation;
+    parsePacket(invitation, xml);
+
+    QCOMPARE(invitation.inviterJid(), "hag66@shakespeare.example");
```

Comment:
You need to use `QCOMPARE(invitation.inviterJid(), QString("hag66@shakespeare.example"))` or `QCOMPARE(invitation.inviterJid(), QStringLiteral("hag66@shakespeare.example"))`. This is causing the compile error on Qt 5.9.